### PR TITLE
8321515: ARM32: Move method resolution information out of the cpCache properly

### PR DESCRIPTION
**Pull request by voitylov**
Thanks to @matias9927, JDK-8320278 fixed the JDK build for ARM32 after JDK-8301997. This PR introduces some additional fixes that enable the ARM32 port to actually work.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8321515](https://bugs.openjdk.org/browse/JDK-8321515): ARM32: Move method resolution information out of the cpCache properly (**Bug** - P2)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17017/head:pull/17017` \
`$ git checkout pull/17017`

Update a local copy of the PR: \
`$ git checkout pull/17017` \
`$ git pull https://git.openjdk.org/jdk.git pull/17017/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17017`

View PR using the GUI difftool: \
`$ git pr show -t 17017`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17017.diff">https://git.openjdk.org/jdk/pull/17017.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17017#issuecomment-1845089729)

### CODE REVIEW

**Comment by bulasevich** [comment]
Thanks for fixing the ARM32 build!
I see the change fixes wrong offset and register usage and does some cleanup. It is OK for me.
Thanks again!

**Comment by voitylov** [comment]
thanks @bulasevich. Any Reviewers, please? I'd like to to get the ARM32 port to work again.

**Comment by shipilev** [comment]
Also changed target-version to 23 in JBS, so that we don't create the backports when we push. You might want to pull and merge the new master to get new mainline with jcheck config for 23.

File: src/hotspot/cpu/arm/interp_masm_arm.cpp [hunk 1]
```diff
@@ -304,16 +304,28 @@ void InterpreterMacroAssembler::load_field_entry(Register cache, Register index,
...
-  mov(cache, sizeof(ResolvedMethodEntry));
-  mul(index, index, cache); // Scale the index to be the entry index * sizeof(ResolvedMethodEntry)
 
-  // load constant pool cache pointer
-  ldr(cache, Address(FP, frame::interpreter_frame_cache_offset * wordSize));
-  // Get address of method entries array
-  ldr(cache, Address(cache, ConstantPoolCache::method_entries_offset()));
-  add(cache, cache, Array<ResolvedMethodEntry>::base_offset_in_bytes());
-  add(cache, cache, index);
+  if (is_power_of_2(sizeof(ResolvedMethodEntry))) {
```
[the remark this answers is not in the document]
Comment (voitylov) [reply src/hotspot/cpu/arm/interp_masm_arm.cpp]: Yes, it's gone in the new version.

**Review by shipilev — APPROVED** [review]
Still good.

**Comment by matias9927** [comment]
Thanks for this fix @voitylov!

**Pull request closed without merging**